### PR TITLE
#36/research about open api as a way to define our api

### LINKED ISSUE #36: Research about OpenApi as a way to define our API

### PR DESCRIPTION
**Pull request by gbudau**
This PR adds the swagger module, which generates OpenAPI documentation automatically.
We can access the documentation at http://localhost:3000/api, and the JSON file at http://localhost:3000/api-json

closes #36

### CODE REVIEW

File: transcendence-app/src/oauth42/oauth42.controller.ts [hunk 1]
```diff
@@ -1,11 +1,15 @@
...
+import { ApiBadGatewayResponse, ApiResponse, ApiTags } from '@nestjs/swagger';
 import { User } from '../user/decorators/user.decorator';
 import { UserDto } from '../user/dto/user.dto';
 import { OAuth42Guard } from './oauth42.guard';
 
+@ApiTags('oauth42')
 @Controller('oauth42')
 export class OAuth42Controller {
   @Get()
+  @ApiResponse({ status: HttpStatus.FOUND })
```
Comment (gbudau) [inline transcendence-app/src/oauth42/oauth42.controller.ts:11]: Here we could use the `@ApiFoundResponse()` decorator

File: transcendence-app/src/oauth42/oauth42.controller.ts [hunk 2]
```diff
@@ -1,11 +1,19 @@
 import { Controller, Get, UseGuards } from '@nestjs/common';
+import {
+  ApiBadGatewayResponse,
+  ApiFoundResponse,
+  ApiTags,
+} from '@nestjs/swagger';
```
Comment (vdedios) [inline transcendence-app/src/oauth42/oauth42.controller.ts:6]: Do we need to add this tags in every controller right?

**Review by vdedios — APPROVED** [review]
LGTM 🚀

File: transcendence-app/package.json [hunk 3]
```diff
@@ -28,13 +28,15 @@
     "@nestjs/core": "^8.0.0",
     "@nestjs/passport": "^8.2.1",
     "@nestjs/platform-express": "^8.0.0",
+    "@nestjs/swagger": "^5.2.1",
```
Comment (Damrod) [inline transcendence-app/package.json:31]: How do you establish what's the version we need/want?

**Review by Damrod — APPROVED** [review]
LGTM

File: transcendence-app/src/oauth42/oauth42.controller.ts (hunk 2 above)
Comment (gbudau) [reply transcendence-app/src/oauth42/oauth42.controller.ts:6]: Correct, with the `ApiTags` decorator, we can group routes. If we don't add this decorator, they get added to the `default` tag.

And if the route has more than one response type, we need to add the decorators for them (if we want those response types to appear in SwaggerUI).

**Review by fizcamposancos — APPROVED** [review]
Good job!

**Comment by gbudau** [comment]
Thank you!

**Pull request merged**